Web/Edubase.Services/IntegrationEndPoints/AzureMaps/AzureMapsService.cs

```diff
@@ -26,12 +26,34 @@ public class AzureMapsService : IAzureMapsService
 
         private static Policy CreateRetryPolicy()
         {
-            var retryIntervalSettings = ConfigurationManager.AppSettings["AzureMapService_RetryIntervals"].Split(',');
-            var retryIntervals = retryIntervalSettings.Select(int.Parse).ToArray();
+            var retryIntervalSettings = ConfigurationManager.AppSettings["AzureMapService_RetryIntervals"];
+
+            if (string.IsNullOrEmpty(retryIntervalSettings?.Trim()))
+            {
+                return Policy.NoOp();
+            }
+
+            var retryIntervals = CsvSecondsToTimeSpans(retryIntervalSettings);
 
             return Policy
                 .Handle<HttpRequestException>()
-                .WaitAndRetryAsync(retryIntervals.Select(seconds => TimeSpan.FromSeconds(seconds)));
+                .WaitAndRetryAsync(retryIntervals);
+        }
+
+        private static TimeSpan[] CsvSecondsToTimeSpans(string csvSeconds)
+        {
+            var retryIntervals = csvSeconds.Split(',')
+                .Select(x => x.Trim())
+                .Where(x => int.TryParse(x, out var seconds) && seconds >= 0)
+                .Select(x => TimeSpan.FromSeconds(int.Parse(x)))
+                .ToArray();
+
+            if (retryIntervals.Length == 0)
+            {
+                return new[] { TimeSpan.FromSeconds(2) };
+            }
```

Comment:
For what reason are we defaulting to two attempts, with 2-seconds between the attempts?

Apologies if I've missed something with this!

Reply:
we shouldn't be - this should be applied to default timeouts not retries